pkg/apis/camel/v1alpha1/kamelet_binding_types.go

```diff
@@ -134,3 +136,21 @@ type KameletBindingList struct {
 	metav1.ListMeta `json:"metadata,omitempty"`
 	Items           []KameletBinding `json:"items"`
 }
+
+// ObjectKinds return the GKV of the KameletBinding spec
+func (kb *KameletBinding) ObjectKinds(runtime.Object) ([]schema.GroupVersionKind, bool, error) {
+	return []schema.GroupVersionKind{
+		{
+			Group:   SchemeGroupVersion.Group,
+			Version: SchemeGroupVersion.Version,
+			Kind:    KameletBindingKind,
+		},
+	}, true, nil
+}
+
+// Recognizes returns true if it match the expected GKV
+func (kb *KameletBinding) Recognizes(schema schema.GroupVersionKind) bool {
+	return schema.Group == SchemeGroupVersion.Group &&
+		schema.Kind == KameletBindingKind &&
+		schema.Version == SchemeGroupVersion.Version
+}
```

Comment:
@squakez why are they needed? (still trying to understand why #2582 CI fails)

Reply:
@nicolaferraro that's the implementation for `runtime.ObjectTyper`, but that's supposed to be implemented by the scheme, into which the `KameletBinding` type should be registered.

Reply:
Yeah, I had to implement those methods as I was using `printer := printers.NewTypeSetter(binding)` where binding was a `*v1alpha1.KameletBinding`. According to the other comment @astefanutti did previously, this would be no longer needed if I use `scheme.Scheme`. Would it be the right way to proceed?

Reply:
Should be fixed now